client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +12,79 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+enum UseFor {
+	IFrameDomain,
+	PreviewDomain,
+}
+
+function findBestDomain( domains: Domain[], useFor: UseFor ) {
+	const mostRecentlyRegisteredDomain: {
+		domain: string | null;
+		timestamp: number | null;
+	} = {
+		domain: null,
+		timestamp: null,
+	};
+
+	for ( let i = 0; i < domains.length; i++ ) {
```

Comment:
In case you are wondering I couldn't use a `forEach` loop here (it would have been cleaner) because on `line 35` where I am returning the `.wordpress.com` domain I want to also return from the `findBestDomain` function as well. If I were to use a `forEach` it would only return from the `forEach` function and not the `findBestDomain` function